shell/platform/linux/fl_platform_views_plugin_test.cc

```diff
@@ -0,0 +1,203 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/linux/fl_platform_views_plugin.h"
+
+#include "gtest/gtest.h"
+
+#include "flutter/shell/platform/linux/fl_binary_messenger_private.h"
+#include "flutter/shell/platform/linux/fl_engine_private.h"
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_method_channel.h"
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_standard_message_codec.h"
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_standard_method_codec.h"
+#include "flutter/shell/platform/linux/testing/fl_test.h"
+#include "flutter/shell/platform/linux/testing/mock_platform_views.h"
+
+static void invoke_response_cb(GObject* source_object,
+                               GAsyncResult* res,
+                               gpointer user_data) {
+  g_main_loop_quit(reinterpret_cast<GMainLoop*>(user_data));
+}
+
+// Called when a the test engine notifies us that a response a sent.
+static void response_cb(FlBinaryMessenger* messenger,
+                        const gchar* channel,
+                        GBytes* message,
+                        FlBinaryMessengerResponseHandle* response_handle,
+                        gpointer user_data) {
+  fl_binary_messenger_send_response(messenger, response_handle, nullptr,
+                                    nullptr);
+
+  g_main_loop_quit(static_cast<GMainLoop*>(user_data));
+}
+
+static void invoke_method(GMainLoop* loop,
+                          FlBinaryMessenger* messenger,
+                          const gchar* channel_name,
+                          const gchar* method_name,
+                          FlValue* args,
+                          GAsyncReadyCallback cb) {
+  g_autoptr(FlStandardMethodCodec) codec = fl_standard_method_codec_new();
+  g_autoptr(FlMethodChannel) channel = fl_method_channel_new(
+      messenger, "test/standard-method", FL_METHOD_CODEC(codec));
+  g_autoptr(FlValue) invoke_args = fl_value_new_list();
+  fl_value_append_take(invoke_args, fl_value_new_string(channel_name));
+  fl_value_append_take(invoke_args, fl_value_new_string(method_name));
+  fl_value_append_take(invoke_args, args);
+  fl_method_channel_invoke_method(channel, "InvokeMethod", invoke_args, nullptr,
+                                  cb, loop);
+}
+
+static void create_platform_view(GMainLoop* loop,
+                                 FlBinaryMessenger* messenger,
+                                 int64_t view_id,
+                                 const gchar* view_type,
+                                 FlValue* args,
+                                 GAsyncReadyCallback cb) {
+  FlValue* create_args = fl_value_new_map();
```

Comment:
Missing g_autoptr

Reply:
`invoke_method` takes ownership of `create_args` here. So we cannot free here.